deployments.go

```diff
@@ -57,18 +59,21 @@ func (s *deployerService) deploy(ctx context.Context, db *gorm.DB, opts DeployOp
 		Config:      config,
 		Slug:        slug,
 		Description: desc,
-	})
+	}, ss)
 
 	return r, err
 }
 
 // Deploy is a thin wrapper around deploy to that adds the error to the
 // jsonmessage stream.
-func (s *deployerService) Deploy(ctx context.Context, db *gorm.DB, opts DeployOpts) (*Release, error) {
+func (s *deployerService) Deploy(ctx context.Context, opts DeployOpts) (*Release, error) {
 	var msg jsonmessage.JSONMessage
 
-	r, err := s.deploy(ctx, db, opts)
+	tx := s.db.Begin()
```

Comment:
i was rolling back here: https://github.com/remind101/empire/pull/917/files#diff-ff47559c0383dd1959cdbe50cd2c036dR76 which is roughly equivalent, but +1 on the separation, i was thinking the same